packages/cli/src/s3-store.ts

```diff
@@ -66,22 +67,51 @@ class S3StoreMap {
 }
 
 export class S3Store implements IKVStore {
-  readonly #storeMap: S3StoreMap
+  readonly #bucketName: string
+  readonly #customEndpoint?: string
+  readonly #diagnosticsLogger: DiagnosticsLogger
+  #storeMap: S3StoreMap
 
   readonly #loadingLimit = new PQueue({
     intervalCap: MAX_LOAD_RPS,
     interval: 1000,
     carryoverConcurrencyCount: true,
   })
 
-  constructor(networkName: string, bucketName: string, customEndpoint?: string) {
+  constructor(
+    networkName: string,
+    diagnosticsLogger: DiagnosticsLogger,
```

Comment:
minor point but if we put this at the end and made it optional we could be backwards compatible

Reply:
AFAIK it's internal to the CLI so I don't think it affects external consumers, can you confirm please @stbrody?

Reply:
yep, no one should be depending on this, seems fine to change